imports/plugins/included/search-mongo/server/methods/searchcollections.js

```diff
@@ -216,7 +216,7 @@ export function buildAccountSearchRecord(accountId, updatedFields) {
     // forceIndex is included to forceIndexing on startup, or when manually added
     const searchableFields = ["forceIndex", "shopId", "emails", "firstName", "lastName", "phone"];
 
-    const shouldRunIndex = updatedFields && updatedFields.some((field) => searchableFields.includes(field));
+    const shouldRunIndex = !Array.isArray(updatedFields) || updatedFields.some((field) => searchableFields.includes(field));
```

Comment:
Can you explain what we are doing here? It seems like we switched from checking if `updatedFields` is an array, to checking if it's not? Why?

Just trying to make sure I have it correct:
Before: If `updatedFields` exists and is an Array (`check` checks this) and includes one of the fields, `shouldRunIndex === true`

After: If `updatedFields` is not an `array` (is that possible at this point? doesn't `check` make sure it is an `array`?) or it is an array and includes one of the fields, `shouldRunIndex === true`

Reply:
This is one of those changes I made months ago when I started this, so I don't remember it that well. But I think I changed this intentionally because this function is called in a `afterAccountUpdate` event consumer function, and I believe sometimes those events do not pass the `updatedFields` property now, or at least we can't rely on them to. My thinking was that if we don't know which fields were updated (i.e., if `updatedFields` isn't an array), we should reindex to be safe.

You make a good point about the `check` above, though. I didn't see that. I'll make it optional so that we actually get to this point when `updatedFields` is undefined. I'll also add a comment here to explain what I just said.